### PR TITLE
Delay initialization of the visual shader editor UI

### LINKED ISSUE #63516: Vector4i and Projection are missing icons
### Godot version

v4.0.alpha.custom_build [5b6135291]

### System information

Windows 10

### Issue description

The new Vector4i and Projection types are missing editor icons.

image
image

### Steps to reproduce

* Press F1
* Search for Vector4i and Projection

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by YuriSizov**
This makes sure that the theme is ready to be used before the method is called. Currently editor spams errors when opening the visual shader editor.

I also noticed some of the icons were missing in the shader editor, which led me to https://github.com/godotengine/godot/issues/63516. So I incorporated all suggested changes, and then some:

* Updated colors and shapes for the vec4 types.
* Added uint and Projection icons.
* Changed the text in Transform3D icon to match Transform2D.
* Cleaned-up submitted icons and added colors to the color conversion map.

Icons in practice:

[godot windows editor dev x86_64_2023-05-23_14-38-39](https://github.com/godotengine/godot/assets/11782833/f4f506ef-0397-4d37-98b3-9c1d08564705)
[godot windows editor dev x86_64_2023-05-23_14-39-04](https://github.com/godotengine/godot/assets/11782833/a4892ab1-8142-4c85-bdc8-1ec26f4a5c16)
[godot windows editor dev x86_64_2023-05-23_14-39-16](https://github.com/godotengine/godot/assets/11782833/91abc509-0ff9-47b3-ba37-c869deb8f5c7)
[godot windows editor dev x86_64_2023-05-23_14-39-46](https://github.com/godotengine/godot/assets/11782833/d0e197c8-6fa0-49de-94dd-fcb34f0f935a)

Closes https://github.com/godotengine/godot/issues/63516.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

File: editor/plugins/visual_shader_editor_plugin.cpp [hunk 1]
```diff
@@ -1845,31 +1845,31 @@ void VisualShaderEditor::_update_options_menu() {
 		}
 		switch (options[i].return_type) {
 			case VisualShaderNode::PORT_TYPE_SCALAR:
-				item->set_icon(0, EditorNode::get_singleton()->get_gui_base()->get_theme_icon(SNAME("float"), SNAME("EditorIcons")));
+				item->set_icon(0, get_theme_icon(SNAME("float"), SNAME("EditorIcons")));
```
[the remark this answers is not in the document]
Comment (YuriSizov) [reply editor/plugins/visual_shader_editor_plugin.cpp:1848]: I started to remove them, but I'm not entirely sure about the lifecycle of those nodes. They are GUI nodes, so technically they can access the theme directly, but most of those are called only from some `setup` method, so I'm concerned about it being called before the node being added to the tree.

This file definitely could use a pass, but I decided to not do it right now.

> Or what defines whether it should be used or not?

If it's done properly, i.e. if it's implemented to react to theme changes, it should never be used like that. An exception can be made when this is needed by a non-GUI node.

**Comment by YuriSizov** [comment]
> This also needs to be updated to use the new icon

Fixed that, should be good to go now. Seems that vec4 icons are also slightly different in that submission from https://github.com/godotengine/godot/issues/63516. I didn't notice it at first, but I actually like the change, so let's keep it.

**Review by AThousandShips — APPROVED** [review]
LGTM, thank you for finally making the icons complete! Discovered both this and Vector4i were missing working on a PR but forgot to open an issue about it (just left the comments) but fortunately there was one already and lovely icons were made

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**